src/value.h

```diff
@@ -12,6 +12,10 @@
 #include <stdio.h>
 #include "./rmutil/vector.h"
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
```

Comment:
These header changes will be necessary in every header file included by unit tests so that the linker can handle object files produced by gcc/clang as well as those produce by g++/clang++.

This could probably be done in a way that doesn't make it local to each header file, but I don't know if that would be bad practice or inadvisable in some way! New territory for me.

This change does not force source files to be re-compiled for `make test`.

Reply:
I would prefer it if we had an implicit way to add extern C to relevant header files, as you've suggested.
Please consult with @mnunberg on that.

Reply:
As @mnunberg suggested, let's try to wrap includes with extern C block, within the unit test file.